src/bindings/node/node.cc

```diff
@@ -0,0 +1,196 @@
+#include <boost/make_shared.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/shared_ptr.hpp>
+#include <iostream>
+#include <node_api.h>
+#include <sstream>
+#include <string>
+
+#include "valhalla/tyr/actor.h"
+
+boost::property_tree::ptree json_to_pt(const char* json) {
+  std::stringstream ss;
+  ss << json;
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(ss, pt);
+  return pt;
+}
+
+boost::property_tree::ptree make_conf(const char* config) {
+  return json_to_pt(config);
+}
+
+// check if napi_status is nap_ok, throw error if not
+void checkNapiStatus(napi_status status, napi_env env, const char* error_message) {
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, error_message);
+  }
+}
+
+#define DECLARE_NAPI_METHOD(name, func)                                                              \
+  { name, 0, func, 0, 0, 0, napi_default, 0 }
+
+class Actor {
```

Comment:
were you going to rename this to `Valhalla` so that its more obvious what it is in js land?

Reply:
I didn't expose the name actor on the js side, so it doesn't really matter what we call it - I think we may as well stick with Actor so it is clear what we're wrapping.

You can see how its used in js-land in the test here- https://github.com/valhalla/valhalla/pull/1341/files#diff-ec6e8fd88a03aee46871535339aed9a8R4